### PR TITLE
6213 Теребилкин Д.И. Лаб.5 Вар.9

### PR DESCRIPTION
**Pull request by Quyntrd**
Честно говоря, я все еще пытаюсь понять, что я вообще делаю, каждый раз запуская ipynb, я как будто бы стреляю себе в колено с ружья 12 калибра. Я пока пытаюсь протестировать разные выборки датасетов, чтобы он работать начал

### CODE REVIEW

**Review by alxmcs — CHANGES_REQUESTED** [review]
Ну когда что-нибудь заработает, запрашивай ревью

**Review by alxmcs — CHANGES_REQUESTED** [review]
Если бы я получил такие результаты, у меня возникло бы подозрение насчет того, все ли я делаю правильно:
image


Твоя задача - провести 9 экспериментов по обучению сети. В результате каждого эксперимента у тебя должен получиться отдельный обученный экземпляр модели.
Еще раз - 1 модель, 9 экземпляров, 9 экспериментов.

В текущем варианте кода у тебя 1 модель, 1 ее экземпляр и он обучается 9 раз подряд, накапливая результаты обучения. Поэтому и графики выглядят как кардиограмма трупа к концу экспериментов:
image

**Comment by Quyntrd** [comment]
Я поменял расположение создания модели, теперь она создается на каждом цикле при тестировании разных learning rate и batch size, и сейчас графики имеют нормальный вид. (Я еще до запрашивания реквеста понял что-то не так с обучением модели, но только сейчас понял, что надо было поменять)

**Review by alxmcs — CHANGES_REQUESTED** [review]
Смотрим графики:
image
На обучающей выборке все идет хорошо, на валидационной - сразу все плохо и и лучше не становится. Значит что-то не так с данными:
Вот тут у тебя есть шаффл, это здоровая идея - перемешивать выборки при обучении:
image
Только вот при разделении на обучающую, валидационную и тестовую выборки шаффла не было:
image
А датасет у тебя сортированный  по меткам:
image
Соответственно, при делении датасета на выборки в обучающую упали только метки 0, а в валидационную - только метки 1 (тут несколько утрировано, но, думаю, идею ты понял). А для качественного обучения модели во всех выборках должны быть представлены все классы в равной (+/-) степени.

**Review by alxmcs — COMMENTED** [review]
У тебя, судя по данным, получается, бинарная классификация отзывов - отзывы на 4 звезды ты пробуешь отличать от 5 звезд (если я верно понял). 
Звучит тяжеловато. Отзывы на 1 звезд от 5 звезд было бы пробовать отличить реалистичнее, там будет явно разный вокабуляр.

Но я не заставляю переделывать датасет, просто держу в курсе

**Comment by Quyntrd** [comment]
Я на самом деле думаю, что это происходит из-за датасета, возможно лучше будет разделить таким образом, отзывы с 4 и 5 звездами ставить им класс 1, а отзывам с 1-3 звездами класс 0, тогда он будет отличать отзывы, гипотетически на хороший/плохой

**Review by alxmcs — APPROVED** [review]
Я тут освежил себе память по документации sklearn и понял, что зря быканул - train_test_split по дефолту перемешивает выборки
